lib/shoryuken/logging.rb

```diff
@@ -30,7 +30,11 @@ def self.initialize_logger(log_target = STDOUT)
     end
 
     def self.logger
-      @logger || initialize_logger
+      unless defined? @logger
+        @logger = initialize_logger
+      end
```

Comment:
In which version of Ruby do you see this warning?

In that case, if it is defined, but `nil`, it won't `initialize_logger `. That may change the functionality, we may need to check defined and nil (presence) to keep it as is. 

WDYT?

Reply:
All versions of Ruby 2.x - yes if someone may initialise it to nil you may need extra checks.